staging/src/k8s.io/kubectl/pkg/describe/describe.go

```diff
@@ -238,8 +243,8 @@ func describerMap(clientConfig *rest.Config) (map[schema.GroupKind]ResourceDescr
 
 // DescriberFor returns the default describe functions for each of the standard
 // Kubernetes types.
-func DescriberFor(kind schema.GroupKind, clientConfig *rest.Config) (ResourceDescriber, bool) {
-	describers, err := describerMap(clientConfig)
+func DescriberFor(kind schema.GroupKind, clientConfig *rest.Config, httpClient *http.Client) (ResourceDescriber, bool) {
```

Comment:
Similar to a previous comment I made, I'm not sure how important it is to maintain the signatures of public functions, or if we want to still support `DescribeFor` that creates its own HTTP client and add a new `DescribeForConfigAndClient` that accepts a client.

I'm not saying it definitely needs to be a separate function, but just asking the question

Reply:
I am also not sure how much the backwards compatibility is important but I have added methods with `AndClient` suffix to be sure. Also applied this to `GenericDescriberFor` as well.